returnn/tf/layers/basic.py

```diff
@@ -3521,18 +3521,26 @@ def get_out_data_from_opts(cls, name, sources, mode="", axes=None, axis=None, ke
       out_size = {
         i if out_batch_dim_axis is None or out_batch_dim_axis >= i else i - 1: size
         for (i, size) in out_size.items()}  # by axis without batch-dim
-    sparse_out = mode.lower().startswith("arg")
-    if sparse_out:
-      out_feature_dim_axis = None
+    if mode == "":
+      # this is for the case that get_out_data_from_opts is called by the SqueezeLayer
```

Comment:
Why? Don't make the code in `ReduceLayer` complicated for a special case by another layer.